### PR TITLE
aws: util: fix extra_user_agent type

### PR DESCRIPTION
**Pull request by PettitWesley**
Fixes: https://github.com/aws/aws-for-fluent-bit/issues/503

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [ ] Attached [local packaging test](./packaging/local-build-all.sh) output showing all targets (including any new ones) build.

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: src/aws/flb_aws_util.c [hunk 1]
```diff
@@ -239,7 +239,7 @@ void flb_aws_client_destroy(struct flb_aws_client *aws_client)
         if (aws_client->upstream) {
             flb_upstream_destroy(aws_client->upstream);
         }
-        if (aws_client->free_user_agent) {
+        if (aws_client->extra_user_agent) {
             /* if user agent was auto-set by code its an SDS string */
```
Comment (PettitWesley) [inline src/aws/flb_aws_util.c:243]: oops forgot to remove the comment

**Comment by PettitWesley** [comment]
## Testing

I used this config, and set and unset the extra_user_agent field:

```
[INPUT]
    Name dummy
    Tag dummy


[OUTPUT]
    Name cloudwatch_logs
    Match   *
    region us-east-1
    log_group_name fluent-bit-cloudwatch
    log_stream_prefix from-fluent-bit-
    auto_create_group On
#    extra_user_agent test-fix
```

This user agent is set in all requests we make, and this is simple fix, so I just ran it under valgrind and checked for output showing invalid memory accesses or leaks. None are shown.

I also set the `FLB_AWS_USER_AGENT` env var. And ran this once without any settings on an ECS instance which should auto-set the user agent to be for ECS. 

A weird DNS error but nothing to worry about from valgrind:
```
[ec2-user@ip-10-192-11-106 build]$ valgrind ./bin/fluent-bit -c ~/user_agent.conf
==25531== Memcheck, a memory error detector
==25531== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==25531== Using Valgrind-3.19.0 and LibVEX; rerun with -h for copyright info
==25531== Command: ./bin/fluent-bit -c /home/ec2-user/user_agent.conf
==25531==
Fluent Bit v1.9.10
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/12/20 00:03:02] [ info] [fluent bit] version=1.9.10, commit=ed21492a94, pid=25531
[2022/12/20 00:03:02] [ info] [storage] version=1.3.0, type=memory-only, sync=normal, checksum=disabled, max_chunks_up=128
[2022/12/20 00:03:02] [ info] [cmetrics] version=0.3.7
[2022/12/20 00:03:03] [ info] [sp] stream processor started
[2022/12/20 00:03:03] [ info] [output:cloudwatch_logs:cloudwatch_logs.0] worker #0 started
==25531== Warning: client switching stacks?  SP change: 0xad0c888 --> 0xa19d480
==25531==          to suppress, use: --max-stackframe=11990024 or greater
==25531== Warning: client switching stacks?  SP change: 0xa19d3f8 --> 0xad0c888
==25531==          to suppress, use: --max-stackframe=11990160 or greater
==25531== Warning: client switching stacks?  SP change: 0xad0c888 --> 0xa19d3f8
==25531==          to suppress, use: --max-stackframe=11990160 or greater
==25531==          further instances of this message will not be shown.
[2022/12/20 00:03:05] [ info] [output:cloudwatch_logs:cloudwatch_logs.0] Creating log stream from-fluent-bit-dummy in log group fluent-bit-cloudwatch
[2022/12/20 00:03:06] [ info] [output:cloudwatch_logs:cloudwatch_logs.0] Log Stream from-fluent-bit-dummy already exists
[2022/12/20 00:03:15] [ warn] [net] getaddrinfo(host='logs.us-east-1.amazonaws.com', err=12): Timeout while contacting DNS servers
[2022/12/20 00:03:15] [error] [aws_client] connection initialization error
^C[2022/12/20 00:03:27] [engine] caught signal (SIGINT)
[2022/12/20 00:03:27] [ warn] [engine] service will shutdown in max 5 seconds
[2022/12/20 00:03:28] [ info] [engine] service has stopped (0 pending tasks)
[2022/12/20 00:03:28] [ info] [output:cloudwatch_logs:cloudwatch_logs.0] thread worker #0 stopping...
[2022/12/20 00:03:28] [ info] [output:cloudwatch_logs:cloudwatch_logs.0] thread worker #0 stopped
==25531==
==25531== HEAP SUMMARY:
==25531==     in use at exit: 110,402 bytes in 3,763 blocks
==25531==   total heap usage: 60,975 allocs, 57,212 frees, 15,915,354 bytes allocated
==25531==
==25531== LEAK SUMMARY:
==25531==    definitely lost: 0 bytes in 0 blocks
==25531==    indirectly lost: 0 bytes in 0 blocks
==25531==      possibly lost: 0 bytes in 0 blocks
==25531==    still reachable: 110,402 bytes in 3,763 blocks
==25531==         suppressed: 0 bytes in 0 blocks
==25531== Rerun with --leak-check=full to see details of leaked memory
==25531==
==25531== For lists of detected and suppressed errors, rerun with: -s
==25531== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

File: include/fluent-bit/flb_aws_util.h [hunk 2]
```diff
@@ -79,7 +79,7 @@ struct flb_aws_client {
     int port;
     char *proxy;
     int flags;
-    flb_sds_t *extra_user_agent;
+    flb_sds_t extra_user_agent;
     int free_user_agent;
```
Comment (matthewfala) [inline include/fluent-bit/flb_aws_util.h:83]: Would you remove this unused variable: `free_user_agent`

File: plugins/out_cloudwatch_logs/cloudwatch_logs.c [hunk 3]
```diff
@@ -333,7 +334,12 @@ static int cb_cloudwatch_init(struct flb_output_instance *ins,
     ctx->cw_client->proxy = NULL;
     ctx->cw_client->static_headers = &content_type_header;
     ctx->cw_client->static_headers_len = 1;
-    ctx->cw_client->extra_user_agent = (char *) ctx->extra_user_agent;
+    tmp_sds = flb_sds_create(ctx->extra_user_agent);
```
Comment (matthewfala) [inline plugins/out_cloudwatch_logs/cloudwatch_logs.c:337]: this simplifies things to let cloudwatch use separate memory for user-agent

**Review by matthewfala — APPROVED** [review]
Looks good! thank you

**Review by matthewfala — CHANGES_REQUESTED** [review]
Would you remove this unused variable: free_user_agent

File: plugins/out_cloudwatch_logs/cloudwatch_logs.c (hunk 3 above)
Comment (PettitWesley) [reply plugins/out_cloudwatch_logs/cloudwatch_logs.c:337]: oops/thanks

Comment (matthewfala) [reply plugins/out_cloudwatch_logs/cloudwatch_logs.c:337]: I think this is a good idea! Just was requesting change on deleting the unused variable free_user_agent

Comment (PettitWesley) [reply plugins/out_cloudwatch_logs/cloudwatch_logs.c:337]: fixed

Comment (PettitWesley) [reply plugins/out_cloudwatch_logs/cloudwatch_logs.c:337]: I meant to reply to the other comment

Comment (PettitWesley) [reply plugins/out_cloudwatch_logs/cloudwatch_logs.c:337]: which I fixed

**Comment by PettitWesley** [comment]
@matthewfala I think this needs to be one of our cherry picks

**Review by matthewfala — APPROVED** [review]

**Pull request merged**